lib/api/bucketPut.js

```diff
@@ -41,7 +41,7 @@ export default function bucketPut(authInfo, request, locationConstraint, log,
     }
     // - AWS JS SDK sends a request with locationConstraint us-east-1
     // if no locationConstraint provided.
-    if (locationConstraint && config.locationConstraints &&
+    if (locationConstraint && !config.regions && config.locationConstraints &&
```

Comment:
@jonathan-gramain, this is more about checking to make sure we do not have the legacy config.